### PR TITLE
External model typed constructor changes

### PR DESCRIPTION
**Pull request by sima-zhu**

### CODE REVIEW

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchAsyncClient.java [hunk 1]
```diff
@@ -189,14 +188,13 @@ HttpPipeline getHttpPipeline() {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<IndexDocumentsResult>> uploadDocumentsWithResponse(Iterable<?> documents,
-        IndexDocumentsOptions options, RequestOptions requestOptions) {
-        return withContext(context -> uploadDocumentsWithResponse(documents, options, requestOptions, context));
+        RequestOptions requestOptions) {
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchAsyncClient.java:191]: You removed `IndexDocumentOptions`

File: sdk/search/azure-search-documents/swagger/readme.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ To see additional help and options, run:
 Fork and clone https://github.com/Azure/autorest.java 
 git checkout v4
 git submodule update --init --recursive
-mvn package -Dlocal
+mvn package -Dlocal -DskipTests
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/swagger/readme.md:22]: Looks like a typo here?

File: sdk/search/azure-search-documents/swagger/readme.md [hunk 3]
```diff
@@ -55,7 +55,7 @@ These settings apply only when `--tag=package-2020-06-searchservice` is specifie
 ``` yaml $(tag) == 'package-2020-06-searchservice'
 namespace: com.azure.search.documents.indexes
 input-file:
-- https://raw.githubusercontent.com/Azure/azure-rest-api-specs/f487ebc93065fd42540aa422c04edd6a768a03e6/specification/search/data-plane/Azure.Search/preview/2020-06-30/searchservice.json
+- https://raw.githubusercontent.com/Azure/azure-rest-api-specs/0bc7853cb4d824bb6c310344dcc1b5f77cbe6bdd/specification/search/data-plane/Azure.Search/preview/2020-06-30/searchservice.json
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/swagger/readme.md:58]: Any reason why we are targeting a commit?

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchAsyncClient.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchAsyncClient.java:191]: This goes for all document indexing APIs.

Not sure how tests aren't failing due to this.

File: sdk/search/azure-search-documents/swagger/readme.md (hunk 2 above)
Comment (sima-zhu) [reply sdk/search/azure-search-documents/swagger/readme.md:22]: The test needs to setup local server to run. So I skip tests which is enough for building the package.

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchAsyncClient.java (hunk 1 above)
Comment (sima-zhu) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchAsyncClient.java:191]: Will double check. That's a miss after merge conflicts.

File: sdk/search/azure-search-documents/swagger/readme.md (hunk 3 above)
Comment (sima-zhu) [reply sdk/search/azure-search-documents/swagger/readme.md:58]: This agrees all across languages. 
The swagger could be changed in future. We don't want to introduce any new changes without notice when regenerate codes.

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchAsyncClient.java (hunk 1 above)
Comment (sima-zhu) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchAsyncClient.java:191]: Do you have tests on throwOnError to false?

Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchAsyncClient.java:191]: There are a few test that are setting the value, don't remember if it was true or false, but it should trigger a compile error.

Comment (sima-zhu) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchAsyncClient.java:191]: Will add tests for the senarios.

**Pull request merged**